src/corelibs/U2Gui/src/PasteController.cpp

```diff
@@ -106,18 +106,19 @@ PasteTask *PasteFactoryImpl::createPasteTask(bool isAddToProject) {
     if (mimeData->hasUrls()) {
         return new PasteUrlsTask(mimeData->urls(), isAddToProject);
     }
-    QString clipboardText;
     try {
-        clipboardText = clipboard->text();
+        QString clipboardText = mimeData->hasFormat(U2Clipboard::UGENE_MIME_TYPE)
+                                    ? QString::fromUtf8(mimeData->data(U2Clipboard::UGENE_MIME_TYPE))
+                                    : clipboard->text();
+        if (clipboardText.isEmpty()) {
+            coreLog.error("UGENE can not recognize current clipboard content as one of the supported formats.");
+            return nullptr;
+        }
+        return new PasteTextTask(clipboardText, isAddToProject);
```

Comment:
OK, not at all, `new` are able to throw `bad_alloc`, but I think it's OK if we don't take it an account.